### PR TITLE
mon/OSDMonitor: make 'osd crush class rename' idempotent

### PR DESCRIPTION
**Pull request by xiexingguo**
Signed-off-by: xie xingguo <xie.xingguo@zte.com.cn>

### CODE REVIEW

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -7609,16 +7609,11 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
...
     CrushWrapper newcrush;
     _get_pending_crush(newcrush);
-
-    if (!newcrush.class_exists(srcname)) {
-      err = -ENOENT;
-      ss << "class '" << srcname << "' does not exist";
-      goto reply;
-    }
-
-    if (newcrush.class_exists(dstname)) {
```
Comment (liewegas) [inline src/mon/OSDMonitor.cc]: should we keep this, so that if both src and dest exist, you still get EEXIST?

Comment (xiexingguo) [reply src/mon/OSDMonitor.cc]: `CrushWrapper::rename_class` will do this:

```c++
int CrushWrapper::rename_class(const string& srcname, const string& dstname)
{
  auto i = class_rname.find(srcname);
  if (i == class_rname.end())
    return -ENOENT;
  auto j = class_rname.find(dstname);
  if (j != class_rname.end())
    return -EEXIST;
```

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-08-29_05:18:12-rados-wip-sage-testing-20170828-distro-basic-smithi/

**Comment by liewegas** [comment]
https://github.com/ceph/ceph/pull/17344

**Pull request merged**